김현국/app/src/main/java/com/example/myapplication/view/search/SearchFragment.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.myapplication.view.search
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.myapplication.R
+import com.example.myapplication.adapter.RepoAdapter
+import com.example.myapplication.base.BaseFragment
+import com.example.myapplication.databinding.FragmentSearchBinding
+import com.example.myapplication.presenter.search.SearchContract
+import com.example.myapplication.presenter.search.SearchPresenter
+/**
+ * @author 김현국
+ * @created 2022/04/25
+ */
+class SearchFragment :
+    BaseFragment<FragmentSearchBinding>(FragmentSearchBinding::inflate),
+    SearchContract.FragmentView {
+
+    private lateinit var searchPresenter: SearchPresenter
+    private lateinit var repoAdapter: RepoAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        searchPresenter.takeView(this)
+        initUI(this.requireContext())
+        arguments?.let {
+            if (it.getSerializable("q") != null) {
+                val q = it.getString("q")
+                if (!q.isNullOrBlank())
+                    searchPresenter.getRepoList(q)
+            }
+        }
+    }
+
+    private fun initUI(context: Context) {
+        repoAdapter = RepoAdapter()
+        binding.rvSearch.layoutManager = LinearLayoutManager(context)
+        binding.rvSearch.adapter = repoAdapter
+
+        searchPresenter.setRepoAdapterModel(repoAdapter)
+        searchPresenter.setRepoAdapterView(repoAdapter)
+
+        searchPresenter.getRepoList()
+    }
```

Comment:
외부에서 호출하는 함수가 아니라서 저는 context를 넣어줄 필요는 없을 것 같아요!
```suggestion
    private fun initUI() {
        repoAdapter = RepoAdapter()
        binding.rvSearch.layoutManager = LinearLayoutManager(requireContext())
        binding.rvSearch.adapter = repoAdapter

        searchPresenter.setRepoAdapterModel(repoAdapter)
        searchPresenter.setRepoAdapterView(repoAdapter)

        searchPresenter.getRepoList()
    }
```